src/text/font.rs

```diff
@@ -95,24 +94,23 @@ impl FontMetrics {
     }
 }
 
+#[self_referencing]
 pub(crate) struct Font {
-    data: Vec<u8>,
-    owned_ttf_font: OwnedFace,
+    data: Box<dyn AsRef<[u8]>>,
+    #[borrows(data)]
+    #[covariant]
+    face_data: rustybuzz::Face<'this>,
     units_per_em: u16,
     metrics: FontMetrics,
     glyphs: FnvHashMap<u16, Glyph>,
 }
 
 impl Font {
-    pub fn new(data: &[u8]) -> Result<Self, ErrorKind> {
-        let owned_ttf_font = OwnedFace::from_vec(data.to_owned(), 0).map_err(|_| ErrorKind::FontParseError)?;
+    pub fn new_with_data<T: AsRef<[u8]> + 'static>(data: T, face_index: u32) -> Result<Self, ErrorKind> {
```

Comment:
I wonder if you would take a Box<dyn AsRef<u8>>  there, if that would work to pass a Box<[u8]> that can be created from a Vec<u8> without allocations.

Reply:
Interesting idea! How exactly would such a `Box<[u8]>` be created from a Vec without allocations?

Reply:
I think I see now. There's a double allocation, we essentially store a `Box<Vec<u8>>`, which is silly. As long as the field member is `data: Box<dyn AsRef<[u8]>>` then I can't seem to cast a `Box<[u8]>` to that though.

However what is possible to make `Font<T>` and store `data` plainly - and instead replace `generational_area:: Arena <Font>` with `generational_area::Arena<Box<dyn FontAccess>>`.

That would change effectively the pattern of `Vec<Font>` where `struct Font { data: Box<dyn AsRef<[u8]>>, other fields}` to `Vec<Box<dyn FontAccess>>` and the `Font` allocation would be better(ish).

I'm also not sure it's worth it though.

Reply:
I think in total there's a triple allocation, `ouroboros` needs to box the `struct Font` for `Pin`.

Reply:
Fair point. While I don't see the entire `struct Font` being pin-boxed (only for the async code path), I can see that all borrowed fields are boxed (with `aliasable`'s `UniqueBox`) - our `data` field.

I'll land this change and then do a follow-up to reduce the allocations.